lesson10/main.js

```diff
@@ -0,0 +1,52 @@
+const ul = document.getElementById("js-ul");
+const loading = document.getElementById("js-loading")
+
+const addLoading = () =>{
+  const loadingImg = document.createElement('img');
+  loadingImg.src = 'loading.gif';
+  loadingImg.id = "loading-img";
+  loading.appendChild(loadingImg);
+};
+
+const removeLoading = () =>{
+  document.getElementById('loading-img').remove();
+};
+
+const createLists = (features) =>{
```

Comment:
IMO
I have had it pointed out to me that 'render' is more appropriate than 'create' for the function here. Because we are not only creating elements, we are inserting and assembling them.

Reply:
This was a very good feedback. I have tried several refactors to make it work. 
Firstly, I simply changed the variable name as your suggestion, then tried to passed a value(which is the feature array) of asyncLoading to the function to render lists. However, I got an error saying "feature is not iteratable". 
I figured that the passed result should be stored in a variable, then being passed to the function to create lists and render it. Therefore, I created a new function called renderLists as your suggestion to handle await loading image to be rendered and remove, then render lists. 
I am not sure this the best way to handle those process tho, but please let me know if there is any other better ways to handle those process. 

Please check my latest commit

Reply:
Q
Is the renderList() you are passing to resolve a different function than renderLists(features)?
If it's not the same thing, it would be great if you could show us the renderList() code too!
I thought that if they were the same thing, then `features` would not need an argument since it is declared in the function.

Reply:
I think it is possible to write the following.
```JavaScript
const getData = new Promise((resolve) => {
  setTimeout(
    () =>
      resolve(
        createLists([
          {to: "bookmark.html",img: "1.png",alt: "画像1",text: "ブックマーク"},
          { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
        ])
      ),
    3000
  );
});
```

Reply:
I think the revised code is good!
I think the following function name is appropriate.

createList() → renderLists()
renderList() →init()

[ref](https://codesandbox.io/s/lesson10-forked-lgxl4z?file=/main.js)

Reply:
thank you for the comments Mai-san and Chihiro-san,
Before I made any changed in the local file, I modified the code in sandbox as there are some unclear concept which I still not quite sure. 
At the moment, I followed Mai-san's advice how to pass renderLists() function in promise object. So  it does not have two steps to render lists as my original one, that is why the function name is just renderLists(). 

Please see my [sandbox](https://codesandbox.io/s/lesson10-t6cv4y?file=/main.js)